src/infra/HttpServiceImpl.ts

```diff
@@ -39,34 +43,61 @@ export class HttpServiceImpl implements HttpService {
     url: string,
     queryParams: QueryParams,
     body: QueryParams,
+    clientSdk: Record<string, string>,
     authToken?: string
   ): Promise<T> {
     const sanitizedBodyParams = sanitizeQueryParams(body);
-    return this.fetch(url, queryParams, {
+    return fetch(url, queryParams, {
       method: RequestMethods.POST,
       body: JSON.stringify(sanitizedBodyParams),
       mode: 'cors',
       ...(authToken && { credentials: 'include' }),
       headers: {
+        'Client-SDK': formatAsHttpHeader(clientSdk),
         'Content-Type': 'application/json',
         ...(authToken && { Authorization: `Bearer ${authToken}`}),
       }
     })
       .then(res => res.json());
   }
+}
 
-  /**
-   * Perform a fetch, using the polyfill if needed.
-   */
-  private fetch(
-    url: string,
-    queryParams: QueryParams,
-    reqInit: RequestInit
-  ): Promise<Response> {
-    const urlWithParams = addParamsToURL(url, queryParams);
-    if (typeof(window) !== 'undefined' && window.fetch) {
-      return window.fetch(urlWithParams, reqInit);
-    }
-    return crossFetch(urlWithParams, reqInit);
+/**
+ * Perform a fetch, using the polyfill if needed.
+ */
+function fetch(
+  url: string,
+  queryParams: QueryParams,
+  reqInit: RequestInit
+): Promise<Response> {
+  const urlWithParams = addParamsToURL(url, queryParams);
+  if (typeof(window) !== 'undefined' && window.fetch) {
+    return window.fetch(urlWithParams, reqInit);
   }
+  return crossFetch(urlWithParams, reqInit);
 }
+
+/**
+ * Converts the JSON representing the Client-SDK agents into the format of an HTTP header.
+ * Agents that have an undefined version are excluded from the returned header.
```

Comment:
Here we don't just exclude `undefined` values but *any* falsy value, i.e. null and blank string as well
Do we even let people have undefined values though? The type for `clientSdk` is `Record<string, string>` not `Record<string, string | undefined>`

Reply:
We let them specify undefined values in CustomClientSdk and it seems like they are able to persist without any type check problems in objects of type `ClientSDK` or `Record<string, string>`

Reply:
after updating the docs lgtm!

Reply:
I removed this part of the docs because I moved the check for a falsy value to `getClientSdk`. This way, when we get an object of type ClientSdk, it won't have unexpected `undefined` values either